File: 0883-car-fleet/0883-car-fleet.cpp
class Solution {
public:
    int carFleet(int target, vector<int>& position, vector<int>& speed) {
        vector<pair<int,double>>combined;
        for(int i=0;i<position.size();i++){
            double time = (double)(target - position[i]) / speed[i];
            combined.push_back({position[i],time});
        }
        sort(combined.rbegin(),combined.rend());
        int carFleet =0;
        double lastTime=0;
        for(int i=0;i<combined.size();i++){
            double curTime = combined[i].second;
            if(curTime > lastTime){
                lastTime=curTime;
                carFleet ++;
            } 
        }
        return carFleet;
    }
};

File: 0883-car-fleet/0883-car-fleet.java
class Solution {
    public int carFleet(int target, int[] position, int[] speed) {
        double[] carFleet = new double[target+1];
        for(int i=0;i<speed.length;i++){
            carFleet[position[i]] = (double)(target-position[i])/speed[i];
        }
        int fleets =0;
        double maxSpeed = 0;
        for(int i= target;i>=0;i--){
            if(carFleet[i] > maxSpeed){
                maxSpeed = carFleet[i];
                fleets++;
            }
        }
        return fleets;
    }
}
